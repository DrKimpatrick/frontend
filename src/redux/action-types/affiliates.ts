export enum AffiliateTypes {
  AddMoreInfo = 'Affiliate/AddMoreInfo',
  Errors = 'Affiliate/Errors',
  Loading = 'Affiliate/Loading'
}

// export interface CompanyOrSchool {
//   _id?: string;
//   name: string;
//   address: string;
//   website: string;
//   accountManagerName: string;
//   email: string;
//   phone: string;
//   userId?: string;
//   createdAt?: string;
//   updatedAt?: string;
// }

interface AddMoreInfo {
  type: typeof AffiliateTypes.AddMoreInfo;
  payload: {
    data: any;
  };
}

interface Errors {
  type: typeof AffiliateTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof AffiliateTypes.Loading;
}

export type CompanyActionTypes = AddMoreInfo | Errors | Loading;
