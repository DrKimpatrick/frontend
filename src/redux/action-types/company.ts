export enum CompanyTypes {
  AddCompany = 'Company/AddCompany',
  GetSpecificCompany = 'Company/GetSpecificCompany',
  UpdateCompany = 'Company/UpdateCompany',
  DeleteCompany = 'Company/DeleteCompany',
  Errors = 'Company/Errors',
  Loading = 'Company/Loading'
}

export interface CompanyOrSchool {
  _id?: string;
  name: string;
  address: string;
  website: string;
  accountManagerName: string;
  email: string;
  phone: string;
  userId?: string;
  createdAt?: string;
  updatedAt?: string;
}

interface AddCompany {
  type: typeof CompanyTypes.AddCompany;
  payload: {
    data: CompanyOrSchool;
  };
}

interface Errors {
  type: typeof CompanyTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof CompanyTypes.Loading;
}

export type CompanyActionTypes = AddCompany | Errors | Loading;
