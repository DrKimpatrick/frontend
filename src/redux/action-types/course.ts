export enum CourseTypes {
  AddCourse = 'CourseTypes/AddCourse',
  ListPendingCourse = 'CourseTypes/ListPendingCourse',
  ListAcceptedCourse = 'CourseTypes/ListAcceptedCourse',
  ListDeclinedCourse = 'CourseTypes/ListDeclinedCourse',
  Loading = 'CourseTypes/Loading',
  Errors = 'CourseTypes/Errors',
  NumberOfPendingCourse = 'CourseTypes/NumberOfPendingCourse',
  NumberOfDeclinedCourse = 'CourseTypes/NumberOfDeclinedCourse',
  NumberOfAcceptedCourse = 'CourseTypes/NumberOfAcceptedCourse',
  ListCourseDetail = 'CourseTypes/ListCourseDetail',
  CourseDetailLoading = 'CourseTypes/CourseDetailLoading'
}

export enum CourseStatus {
  Accepted = 'accepted',
  Pending = 'pending',
  Declined = 'declined'
}

export enum CourseTimeFormat {
  Minute = 'Min',
  Hour = 'Hr'
}

export enum CourseLevel {
  Advanced = 'advanced',
  Beginner = 'beginner',
  Intermediate = 'intermediate'
}

export interface Course {
  _id: string;
  currentLangSpecsUpdated: boolean;
  verificationStatus: string;
  name: string;
  instructor: string;
  languageTaught: string;
  existingCourseLink: string;
  coverImageLink: string;
  userId: {
    _id: string;
    username: string;
    email: string;
    avatar?: string;
  };
  createdAt: string;
  updatedAt: string;
  level: string;
  description: string;
  duration: string;
}

export interface CoursePagination {
  offset?: number;
  limit?: number;
}
interface ListAcceptedCourse {
  type: typeof CourseTypes.ListAcceptedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface ListPendingCourse {
  type: typeof CourseTypes.ListPendingCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface ListDeclinedCourse {
  type: typeof CourseTypes.ListDeclinedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}
interface Loading {
  type: typeof CourseTypes.Loading;
  payload: {
    loading: boolean;
  };
}

interface Errors {
  type: typeof CourseTypes.Errors;
  payload: {
    errors: any;
  };
}

interface NumberOfPendingCourse {
  type: typeof CourseTypes.NumberOfPendingCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface NumberOfDeclinedCourse {
  type: typeof CourseTypes.NumberOfDeclinedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface NumberOfAcceptedCourse {
  type: typeof CourseTypes.NumberOfAcceptedCourse;
  payload: {
    data: {
      data: Course[];
      totalItems: number;
      lastUpdatedItem: Course;
    };
  };
}

interface ListCourseDetail {
  type: typeof CourseTypes.ListCourseDetail;
  payload: {
    data: Course;
  };
}

interface CourseDetailLoading {
  type: typeof CourseTypes.CourseDetailLoading;
  payload: {
    loading: boolean;
  };
}

export type CourseActionTypes =
  | ListAcceptedCourse
  | ListDeclinedCourse
  | ListPendingCourse
  | Loading
  | Errors
  | NumberOfAcceptedCourse
  | NumberOfDeclinedCourse
  | NumberOfPendingCourse
  | ListCourseDetail
  | CourseDetailLoading;
