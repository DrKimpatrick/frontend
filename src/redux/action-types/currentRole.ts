export const CURRENT_ROLE_LOADING = 'CURRENT_ROLE_LOADING';
export const CURRENT_ROLE_FAIL = 'CURRENT_ROLE_FAIL';
export const CURRENT_ROLE_SUCCESS = 'FORGOT_SUCCESS';

export interface CurrentRole {
    email: string
}

export interface currentRoleLoading  {
    type: typeof CURRENT_ROLE_LOADING
}

export interface currentRoleFail  {
    type: typeof CURRENT_ROLE_FAIL,
    payload: string
}

export interface currentRoleSuccess  {
    type: typeof CURRENT_ROLE_SUCCESS,
    payload: any
}

export type CurrentRoleDispatchTypes = currentRoleLoading | currentRoleFail | currentRoleSuccess;
