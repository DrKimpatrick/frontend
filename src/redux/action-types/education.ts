export enum EducationTypes {
  AddEducation = 'Education/AddEducation',
  DeleteEducation = 'Education/DeleteEducation',
  UpdateEducation = 'Education/UpdateEducation',
  ListEducation = 'Education/ListEducation',
  ListSpecificEducation = 'Education/ListSpecificEducation',
  Errors = 'Education/Errors',
  Loading = 'Education/Loading',
  ChangeEducationStatus = 'Education/ChangeEducationStatus'
}

export enum VerificationStatus {
  Unverified = 'unverified',
  InProgress = 'inProgress',
  Verified = 'verified'
}
export interface Education {
  _id: string;
  schoolName: string;
  level: string;
  degreeOrCertification: string;
  specializations: string[];
  startDate: string;
  endDate: string;
  accomplishments: string[];
  isCurrentEducation: boolean;
  verificationStatus?: VerificationStatus;
  createdAt?: string;
  updatedAt?: string;
}

interface AddEducation {
  type: typeof EducationTypes.AddEducation;
  payload: {
    data: Education;
  };
}

interface DeleteEducation {
  type: typeof EducationTypes.DeleteEducation;
  payload: {
    id: string;
  };
}

interface UpdateEducation {
  type: typeof EducationTypes.UpdateEducation;
  payload: {
    data: Education;
  };
}

interface ListEducation {
  type: typeof EducationTypes.ListEducation;
  payload: {
    data: Education[];
  };
}

interface ListSpecificEducation {
  type: typeof EducationTypes.ListSpecificEducation;
  payload: {
    data: Education;
  };
}

interface Errors {
  type: typeof EducationTypes.Errors;
  payload: {
    errors: any;
  };
}

interface Loading {
  type: typeof EducationTypes.Loading;
  payload: {
    loading?: boolean;
  };
}

interface ChangeEducationStatus {
  type: typeof EducationTypes.ChangeEducationStatus;
  payload: {
    data: Education;
  };
}

export type EducationActionTypes =
  | AddEducation
  | DeleteEducation
  | UpdateEducation
  | ListEducation
  | ListSpecificEducation
  | Errors
  | Loading
  | ChangeEducationStatus;
