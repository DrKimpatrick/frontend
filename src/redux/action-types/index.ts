// This is where we will add action types