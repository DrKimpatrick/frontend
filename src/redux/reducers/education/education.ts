import { map } from 'lodash';
import { EducationTypes } from '../../actions/education';
import { MessageActionType } from '../../action-types/message';
import { EducationActionTypes, Education } from '../../action-types/education';
import { MessageTypes } from '../../actions/message';

interface InitialState {
  readonly message?: string;
  readonly errors?: any;
  readonly educations?: Education[];
  readonly education?: Education;
  readonly loading?: boolean;
}

const initialState: InitialState = {};

export const educationReducer = (
  state = initialState,
  action: EducationActionTypes | MessageActionType
) => {
  switch (action.type) {
    case EducationTypes.AddEducation:
      return {
        ...state,
        educations: state.educations && [
          ...state.educations,
          action.payload.data
        ],
        loading: false,
        education: action.payload.data
      };

    case EducationTypes.DeleteEducation:
      return {
        ...state,
        loading: false,
        educations:
          state.educations &&
          state.educations.filter(item => item._id !== action.payload.id)
      };

    case EducationTypes.ListEducation:
      return { ...state, loading: false, educations: action.payload.data };

    case EducationTypes.ListSpecificEducation:
      return { ...state, loading: false, education: action.payload.data };

    case EducationTypes.UpdateEducation:
      const update = map(state.educations, item => {
        if (item._id === action.payload.data._id) {
          item = action.payload.data;
        }
        return item;
      });
      return {
        ...state,
        loading: false,
        educations: state.educations && update
      };

    case EducationTypes.Errors:
      return { ...state, loading: false, errors: action.payload.errors };

    case EducationTypes.Loading:
      return {
        ...state,
        loading: action.payload.loading ? action.payload.loading : true
      };

    case MessageTypes.Error:
    case MessageTypes.Message:
      return { ...state, loading: false };

    default:
      return state;
  }
};
