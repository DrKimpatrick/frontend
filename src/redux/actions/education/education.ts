import { Dispatch } from 'redux';
import { InitialEducationValue as InitialValue } from 'components/Talent/Dashboard/Education';
import { EducationTypes } from 'redux/action-types/education';
import { UserTypes } from 'redux/action-types/user';
import { setProfileProcess } from 'redux/actions/user';
import ApiAction from '../../../helpers/apiAction';
import { setMessage } from '../message';

export const addEducation = (
  data: InitialValue,
  userInfo?: { userId: string; profileProcess: string }
) => (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: '/education',
      method: 'POST',
      data,
      onStart: () => (dispatch: Dispatch) =>
        dispatch({ type: EducationTypes.Loading, payload: { loading: true } }),
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: EducationTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        if (userInfo && userInfo.profileProcess && userInfo.userId) {
          setProfileProcess({
            userId: userInfo.userId,
            profileProcess: userInfo.profileProcess
          })(dispatch);
        }
        dispatch({
          type: EducationTypes.AddEducation,
          payload: { data: res.data }
        });
        setMessage('education saved successfully')(dispatch);
      }
    })
  );
};

export const listEducation = () => (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: '/education',
      method: 'GET',
      onStart: () => (dispatch: Dispatch) => {
        dispatch({ type: EducationTypes.Loading, payload: { loading: true } });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: EducationTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
        dispatch({ type: EducationTypes.Loading, payload: { loading: false } });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({ type: EducationTypes.Loading, payload: { loading: false } });
        if (res.data) {
          dispatch({
            type: EducationTypes.ListEducation,
            payload: {
              data: res.data
            }
          });
        }
      }
    })
  );
};

export const listSpecificEducation = (userId: string, educationId: string) => (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: `/education/${userId}/${educationId}`,
      method: 'GET',
      onStart: () => (dispatch: Dispatch) =>
        dispatch({ type: EducationTypes.Loading, payload: { loading: true } }),
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: EducationTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: EducationTypes.ListSpecificEducation,
          payload: { data: res.data }
        });
      }
    })
  );
};

export const updateEducation = (
  data: InitialValue,
  userId: string,
  educationId: string
) => (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: `/education/${userId}/${educationId}`,
      method: 'PATCH',
      data,
      onStart: () => (dispatch: Dispatch) =>
        dispatch({ type: EducationTypes.Loading, payload: { loading: true } }),
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: EducationTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: EducationTypes.UpdateEducation,
          payload: { data: res.data }
        });
        setMessage('education updated successfully')(dispatch);
      }
    })
  );
};

export const deleteEducation = (educationId: string) => (
  dispatchAction: Dispatch
) => {
  return dispatchAction(
    ApiAction({
      url: `/education/${educationId}`,
      method: 'Delete',
      onStart: () => (dispatch: Dispatch) =>
        dispatch({ type: EducationTypes.Loading, payload: { loading: true } }),
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: EducationTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: () => (dispatch: Dispatch) => {
        dispatch({
          type: EducationTypes.DeleteEducation,
          payload: { id: educationId }
        });
        setMessage('education deleted successfully')(dispatch);
      }
    })
  );
};

export const changeEducationStatus = (values: {
  status: string;
  id: string;
}) => (dispatchAction: Dispatch) => {
  return dispatchAction(
    ApiAction({
      url: `/education/status/${values.id}`,
      method: 'Put',
      data: { verificationStatus: values.status },
      onStart: () => dispatch => {
        dispatch({
          type: UserTypes.UserEducationLoading,
          payload: { loading: true }
        });
      },
      onFailure: error => (dispatch: Dispatch) => {
        dispatch({
          type: EducationTypes.Errors,
          payload: {
            errors: error.response.data
          }
        });
      },
      onSuccess: res => (dispatch: Dispatch) => {
        dispatch({
          type: EducationTypes.ChangeEducationStatus,
          payload: { data: res.data }
        });
        setMessage(res.message)(dispatch);
      }
    })
  );
};
