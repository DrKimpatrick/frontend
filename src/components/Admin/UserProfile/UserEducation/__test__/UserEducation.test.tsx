import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { Store } from 'redux';
import { cleanup, render } from '@testing-library/react';
import Component from '../UserEducation';
import { VerificationStatus } from 'redux/action-types/education';
import configureMockStore from 'redux-mock-store';

let store: Store;

const middleware = [thunk];

const mockStore = configureMockStore(middleware);

const education = {
  _id: 'id',
  schoolName: 'companyName',
  verificationStatus: VerificationStatus.Verified,
  startDate: 'startDate',
  endDate: 'endDate',
  specializations: ['specialization'],
  isCurrentEducation: false,
  degreeOrCertification: 'degree',
  accomplishments: ['IT'],
  level: 'level'
};

const props = {
  education: [education],
  userEducationLoading: false
};

describe('UserEducation', () => {
  beforeEach(() => {
    store = mockStore({});
  });
  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <Provider store={store}>
        <Component {...props} />
      </Provider>,
      div
    );
  });

  it('should create snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <Component {...props} />
      </Provider>
    );

    expect(tree).toMatchSnapshot();
  });

  it('should return no result found', () => {
    const { container } = render(
      <Provider store={store}>
        <Component {...props} education={[]} />
      </Provider>
    );

    const message = container.querySelector('.notFound h5');

    expect(message).toHaveTextContent('There are no education');
  });

  it('should return loading spinner', () => {
    const { container } = render(
      <Provider store={store}>
        <Component {...props} userEducationLoading={true} />
      </Provider>
    );

    const loading = container.querySelector('.sideLoading');

    expect(loading).toBeDefined();
  });
});
