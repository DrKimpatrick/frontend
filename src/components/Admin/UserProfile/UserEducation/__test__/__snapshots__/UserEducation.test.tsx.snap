// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEducation should create snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="items educationItem"
      >
        <div
          class="title"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82zM12 3L1 9l11 6 9-4.91V17h2V9L12 3z"
            />
          </svg>
          <h1>
            Education
          </h1>
        </div>
        <div
          class="listItems"
        >
          <ul
            class="p-0 m-2 bg-card-preview"
          >
            <li
              class="py-2 py-2 rounded-sm"
            >
              <h5>
                companyName
              </h5>
              <div
                class="selectItems"
              >
                <select>
                  <option
                    value="verified"
                  >
                    verified
                  </option>
                  <option
                    value="unverified"
                  >
                    unverified
                  </option>
                  <option
                    value="inProgress"
                  >
                    inProgress
                  </option>
                </select>
                <span
                  class="customArrow"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="items educationItem"
    >
      <div
        class="title"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82zM12 3L1 9l11 6 9-4.91V17h2V9L12 3z"
          />
        </svg>
        <h1>
          Education
        </h1>
      </div>
      <div
        class="listItems"
      >
        <ul
          class="p-0 m-2 bg-card-preview"
        >
          <li
            class="py-2 py-2 rounded-sm"
          >
            <h5>
              companyName
            </h5>
            <div
              class="selectItems"
            >
              <select>
                <option
                  value="verified"
                >
                  verified
                </option>
                <option
                  value="unverified"
                >
                  unverified
                </option>
                <option
                  value="inProgress"
                >
                  inProgress
                </option>
              </select>
              <span
                class="customArrow"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
